lib/ui/painting/image_encoding.cc

```diff
@@ -96,6 +81,64 @@ sk_sp<SkImage> ConvertToRasterImageIfNecessary(sk_sp<SkImage> image,
   return snapshot->makeRasterImage();
 }
 
+void ConvertImageToRaster(sk_sp<SkImage> image,
+                          std::function<void(sk_sp<SkImage>)> encode_task,
+                          fml::RefPtr<fml::TaskRunner> gpu_task_runner,
+                          fml::RefPtr<fml::TaskRunner> io_task_runner,
+                          GrContext* resource_context,
+                          fml::WeakPtr<SnapshotDelegate> snapshot_delegate) {
+  // Check validity of the image.
+  if (image == nullptr) {
+    FML_LOG(ERROR) << "Image was null.";
+    encode_task(nullptr);
+    return;
+  }
+
+  auto dimensions = image->dimensions();
+
+  if (dimensions.isEmpty()) {
+    FML_LOG(ERROR) << "Image dimensions were empty.";
+    encode_task(nullptr);
+    return;
+  }
+
+  SkPixmap pixmap;
+  if (image->peekPixels(&pixmap)) {
```

Comment:
Nit: negating `SkImage::isTextureBacked` might be clearer to read. But this is fine too.